### PR TITLE
Add file path to sys.argv when running as module

### PR DESCRIPTION
**Pull request by karthiknadig**
Addresses #617 

@fabioz I am not sure if this is the right fix for the problem. Let me know if I need to make changes in other places.

### CODE REVIEW

**Comment by fabioz** [comment]
Seems like a good fix for me... can you just check what's expected correct when running a directory (containing a `__main__.py`)?

i.e.: just want to know if that assign should be a few lines below or not.

**Comment by karthiknadig** [comment]
@fabioz I had to make a bunch of changes to handle the directory case. This is **NOT** final.
There is a slight difference in behavior depending on if `__init__.py` is present or not.

File: ptvsd/_vendored/pydevd/pydevd.py [hunk 1]
```diff
@@ -974,11 +974,22 @@ def run(self, file, globals=None, locals=None, is_module=False, set_trace=True):
...
+                    if filename is None:
+                        sys.stderr.write("No module named %s\n" % file)
+                        return
+                    else:
+                        file = filename
             else:
                 file = filename
-            sys.argv[0] = file
+                if filename.endswith('__init__.pyc') or filename.endswith('__init__.py'):
+                    filename = filename.replace('__init__.py', '__main__.py')
```
Comment (DonJayamanne) [inline ptvsd/_vendored/pydevd/pydevd.py]: If __main__.pyc exists and not the .py file, then we need to use the pyc counterpart.

**Comment by karthiknadig** [comment]
@fabioz Can you review this?

**Comment by karthiknadig** [comment]
@DonJayamanne Review this please?

**Pull request merged**